olp-cpp-sdk-core/include/olp/core/thread/Continuation.inl

```diff
@@ -29,29 +29,48 @@ class TaskScheduler;
 
 template <typename ResultType>
 void Continuation<ResultType>::Run() {
+  // Run should not start an execution if the `Finally` method is not called
+  // for the task continuation.
   if (!finally_callback_) {
     impl_.Clear();
     return;
   }
 
-  impl_.SetFailedCallback(
-      [this](client::ApiError error) { finally_callback_(std::move(error)); });
-
+  // In case of the continuation is cancelled before calling the `Run` method,
+  // return the `Cancelled` error.
   if (impl_.Cancelled()) {
-    finally_callback_(client::ApiError::Cancelled());
+    const auto finally_callback = std::move(finally_callback_);
+    if (finally_callback) {
+      finally_callback(client::ApiError::Cancelled());
+    }
+
     impl_.Clear();
-    finally_callback_ = nullptr;
     return;
   }
 
-  impl_.Run([this](void* input, bool cancelled) {
-    if (cancelled) {
-      finally_callback_(client::ApiError::Cancelled());
-    } else {
-      auto value = std::move(*static_cast<ResultType*>(input));
-      finally_callback_(std::move(value));
+  // Use std::move to set the `finally_callback_` member to nullptr.
+  // It's needed to check if the execution is done, and prevent calling
+  // the `Cancel` method of the `Continuation` object.
+  // The `SetError` method of the `ExecutionContext` should not execute a
+  // callback as well.
```

Comment:
```suggestion
 /**
   * @brief Sets the `finally_callback_` member to nullptr.
   *
   * It checks if the execution is done and prevents calling
   * the `Cancel` method of the `Continuation` object.
   * The `SetError` method of the `ExecutionContext` object does not
   * execute a callback.
   */
```
Would it be correct to shorten the method names like this: `Continuation::Cancel` ?